### PR TITLE
Adding sari metric

### PR DESCRIPTION
**Pull request by ddhruvkr**
Adding SARI metric that is used in evaluation of text simplification. This is required as part of the GEM benchmark.

### CODE REVIEW

File: metrics/sari/sari.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        )
+
+    def _compute(self, sources, predictions, references):
+        srcs = [normalize(sent) for sent in sources]
+        preds = [normalize(sent) for sent in predictions]
+        refs = [[normalize(sent) for sent in ref_sents] for ref_sents in references]
+
+        sari_score = []
+        for i in range(len(srcs)):
+            sari_score.append(SARIsent(srcs[i], preds[i], refs[i]))
```
Comment (lhoestq) [inline metrics/sari/sari.py:283]: Maybe we could have something that doesn't store the normalized sentences to save memory ?
Also we can add a check that there are as many sources as predictions/references.

Maybe something like that ?

```suggestion
        if not (len(sources) == len(predictions) == len(references)):
            raise ValueError("Sources length must match predictions and references lengths.")

        sari_score = 0
        for src, pred, refs in zip(sources, predictions, references):
            sari_score += SARIsent(
                normalize(src), normalize(pred), [normalize(sent) for sent in refs]
            ))
        sari_score = sari_score / len(predictions)
```

**Review by lhoestq — COMMENTED** [review]
Nice ! Thanks for adding this metric :)

I added a suggestion to make it more memory efficient, let me know what you think

File: metrics/sari/sari.py (hunk 1 above)
Comment (ddhruvkr) [reply metrics/sari/sari.py]: Thanks. I have modified the code based on the suggestions.

**Review by lhoestq — APPROVED** [review]
Nice thanks !

**Pull request merged**